Bytecode verification fails on generated serializer

RecordTest.testNotSerialRecord fails  If  bytecode verification is enabled via "one.nio.gen.verify_bytecode" system property

```
Caused by: java.lang.IllegalArgumentException: Error at instruction 15: Expected an object reference, but found . read(Lone/nio/serial/DataStream;)Ljava/lang/Object;
00000 R R . . . .  :  :     ALOAD 1
00001 R R . . . .  : R  :     NEW one/nio/serial/RecordTest$SomeRecord
00002 R R . . . .  : R R  :     DUP
00003 R R . . . .  : R R R  :     INVOKESPECIAL java/lang/Object.<init> ()V
00004 R R . . . .  : R R  :     DUP_X1
00005 R R . . . .  : R R R  :     INVOKEVIRTUAL one/nio/serial/DataStream.register (Ljava/lang/Object;)V
00006 R R . . . .  : R  :     ALOAD 1
00007 R R . . . .  : R R  :     INVOKEVIRTUAL one/nio/serial/DataStream.readObject ()Ljava/lang/Object;
00008 R R . . . .  : R R  :     CHECKCAST java/lang/String
00009 R R . . . .  : R R  :     ASTORE 3
00010 R R . R . .  : R  :     ALOAD 1
00011 R R . R . .  : R R  :     INVOKEVIRTUAL one/nio/serial/DataStream.readObject ()Ljava/lang/Object;
00012 R R . R . .  : R R  :     POP
00013 R R . R . .  : R  :     DUP
00014 R R . R . .  : R R  :     ALOAD 3
00015 R R . R . .  : R R R  :     ALOAD 5
00016 ?      :     INVOKESPECIAL one/nio/serial/RecordTest$SomeRecord.<init> (Ljava/lang/String;Ljava/lang/String;)V
00017 ?      :     ARETURN
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -127,6 +127,9 @@ tasks {
             languageVersion.set(JavaLanguageVersion.of(testJdk))
         })
 
+
+        systemProperty("one.nio.gen.verify_bytecode", true)
+
         useJUnit()
         testLogging {
             debug {

--- a/src/main/java/one/nio/gen/BytecodeGenerator.java
+++ b/src/main/java/one/nio/gen/BytecodeGenerator.java
@@ -269,6 +269,26 @@ public class BytecodeGenerator extends ClassLoader implements BytecodeGeneratorM
         }
     }
 
+    public static void generateDefault(MethodVisitor mv, Field field) {
+        mv.visitInsn(defaultValueOpcode(Type.getType(field.getType())));
+    }
+
+    public static int defaultValueOpcode(Type type) {
+        switch (type.getSort()) {
+            case Type.OBJECT:
+            case Type.ARRAY:
+                return ACONST_NULL;
+            case Type.FLOAT:
+                return FCONST_0;
+            case Type.DOUBLE:
+                return DCONST_0;
+            case Type.LONG:
+                return LCONST_0;
+            default:
+                return ICONST_0;
+        }
+    }
+
     @Override
     public String getDumpPath() {
         return dumpPath;

--- a/src/main/java/one/nio/serial/gen/DelegateGenerator.java
+++ b/src/main/java/one/nio/serial/gen/DelegateGenerator.java
@@ -273,6 +273,10 @@ public class DelegateGenerator extends BytecodeGenerator {
                 mv.visitVarInsn(ALOAD, 1);
                 mv.visitMethodInsn(INVOKEVIRTUAL, "one/nio/serial/DataStream", srcType.readMethod(), srcType.readSignature(), false);
                 mv.visitInsn(srcType.convertTo(FieldType.Void));
+                if (isRecord) {
+                    generateDefault(mv, ownField);
+                    storeRecordArgument(mv, ownField, fd);
+                }
             } else {
                 if (!isRecord) mv.visitInsn(DUP);
                 if (parentField != null) emitGetField(mv, parentField);
@@ -437,6 +441,10 @@ public class DelegateGenerator extends BytecodeGenerator {
         for (FieldDescriptor fd : fds) {
             Field ownField = fd.ownField();
             if (isNotSerial(ownField)) {
+                if (isRecord) {
+                    generateDefault(mv, ownField);
+                    storeRecordArgument(mv, ownField, fd);
+                }
                 continue;
             }
             fd.next = fieldHashes.put(ownField.getName().hashCode(), fd);
@@ -885,7 +893,7 @@ public class DelegateGenerator extends BytecodeGenerator {
 
     private static void emitPutSerialField(MethodVisitor mv, Field f, boolean isRecord, FieldDescriptor fd) {
         if (isRecord) {
-            mv.visitVarInsn(Type.getType(f.getType()).getOpcode(ISTORE), 3 + fd.index() * 2);
+            storeRecordArgument(mv, f, fd);
             return;
         }
 
@@ -906,6 +914,10 @@ public class DelegateGenerator extends BytecodeGenerator {
         } else {
             emitPutField(mv, f);
         }
+    }
+
+    private static void storeRecordArgument(MethodVisitor mv, Field f, FieldDescriptor fd) {
+        mv.visitVarInsn(Type.getType(f.getType()).getOpcode(ISTORE), 3 + fd.index() * 2);
     }
 
     private static Label emitNullGuard(MethodVisitor mv, Class<?> dst) {

